UnitTestingApp.js

```diff
@@ -52,24 +60,93 @@
       if (console.clear) console.clear();
     }
 
+    getLevelInfo() {
+      return _levelInfo.get(this);
+    }
+
+    setLevelInfo(value) {
+      _levelInfo.set(this, value);
+    }
+
+    /**
+    * Reset statistics counters
+    * @return {void}
+    */
+    resetTestCounters() {
+      _nTests.set(this, 0);
+      _nFailTests.set(this, 0);
+      _nPassTests.set(this, 0);
+    }
+
     /**
      * Tests whether conditions pass or not
      * @param {Boolean | Function} condition - The condition to check
-     * @param {String} message - the message to display in the onsole
+     * @param {String} message - the message to display in the console (based on _levelInfo value)
      * @return {void}
      */
     assert(condition, message) {
-      if(!_enabled.get(this)) return;
-      if(this.isInGas !== this.runningInGas) return;
+      if (!_enabled.get(this)) return;
+      if (this.isInGas !== this.runningInGas) return;
+      _nTests.set(this, _nTests.get(this) + 1);
       try {
         if ("function" === typeof condition) condition = condition();
-        if (condition) console.log(`✔ PASSED: ${message}`);
-        else console.log(`❌ FAILED: ${message}`);
-      } catch(err) {
-        console.log(`❌ FAILED: ${message} (${err})`);
+        if (condition) {
+          _nPassTests.set(this, _nPassTests.get(this) + 1);
+          if (this.getLevelInfo() > 0) console.log(`✔ PASSED: ${message}`);
+        }
+        else {
+          _nFailTests.set(this, _nFailTests.get(this) + 1);
+          if (this.getLevelInfo() > 0) console.log(`❌ FAILED: ${message}`);
+        }
+
+      } catch (err) {
+        _nFailTests.set(this, _nFailTests.get(this) + 1);
+        if (this.getLevelInfo() > 0) console.log(`❌ FAILED: ${message} (${err})`);
+      }
+    }
+
+    /**
+     * Tests whether fun result is equal to expected result or not
+     * @param {Boolean | Function} Condition or fun - to check
+     * @param {String} expectedResult - The expected result to validate
+     * @param {String} message - If present, then used as message to display in the console (based on _levelInfo value). 
+     *                           If message is not present, then in case the result is not equal, it shows the missmatch
+     *                           In the form of: "result != expectedResult". If the result is valid and message is not 
+     *                           provided, then it only indicates the test passed.
+     * @return {void}
+     */
+    assertEquals(fun, expectedResult, message = null) {
+      if (!_enabled.get(this)) return;
+      if (this.isInGas !== this.runningInGas) return;
+      _nTests.set(this, _nTests.get(this) + 1);
+      let msg, result;
+
+      try {
+        // Checking preconditions
+        if (fun === undefined) throw new Error("Provide 'fun' input argument");
+        if (expectedResult === undefined) throw new Error("Provide 'expectedResult' input argument");
```

Comment:
Hmm, I'm now thinking what if we **want** a condition to be `undefined`, I mean that's not crazy, is it?

Reply:
What I'm saying is that I don't think we should be throwing an error when our condition or expected result (or both) are `undefined`. We sometimes **want** them to be `undefined`.